includes/class-wc-calypso-bridge-admin-menu.php

```diff
@@ -10,11 +10,331 @@
+			$args[ 'menu_icon' ]           = 'dashicons-cart';
+			return $args;
+		} );
+	}
+
+	/**
+	 * Groups WooCommerce items.
+	 */
+	public static function menu_order( $menu_order ) {
+
+		// Initialize our custom order array.
+		$woocommerce_menu_order = array();
+
+		// Get the index of our managed pages.
+		$orders    = array_search( 'edit.php?post_type=shop_order', $menu_order, true );
+		$customers = array_search( 'admin.php?page=wc-admin&path=/customers', $menu_order, true );
+		$products  = array_search( 'edit.php?post_type=product', $menu_order, true );
+		$analytics = array_search( 'wc-admin&path=/analytics/overview', $menu_order, true );
+		$marketing = array_search( 'woocommerce-marketing', $menu_order, true );
+		$extensions                = array_search( 'woocommerce', $menu_order, true );
+		$woocommerce_separator     = array_search( 'separator-woocommerce', $menu_order, true );
+		$woocommerce_separator_top = array_search( 'wc-calypso-bridge-separator-top', $menu_order, true );
+		$woocommerce_payments      = array_search( 'wc-admin&path=/payments/connect', $menu_order, true );
+		error_log( "woocommerce_payments:" . $woocommerce_payments );
+		// Loop through menu order and do some rearranging.
+		foreach ( $menu_order as $index => $item ) {
+
+			// Move the WC group above the "Posts" menu item.
+			if ( 'edit.php' === $item ) {
+				$woocommerce_menu_order[] = 'wc-calypso-bridge-separator-top'; // Separator WC top.
+				$woocommerce_menu_order[] = 'edit.php?post_type=shop_order'; // Orders.
+				$woocommerce_menu_order[] = 'edit.php?post_type=product'; // Products.
+				$woocommerce_menu_order[] = 'admin.php?page=wc-admin&path=/customers'; // Customers.
+				if ( false !== $woocommerce_payments ) {
+					$woocommerce_menu_order[] = 'wc-admin&path=/payments/connect'; // Payments.
+				}
+
+				$woocommerce_menu_order[] = 'wc-admin&path=/analytics/overview'; // Analytics.
+				$woocommerce_menu_order[] = 'woocommerce-marketing'; // Marketing.
+				$woocommerce_menu_order[] = 'woocommerce'; // Extensions.
+				$woocommerce_menu_order[] = 'separator-woocommerce'; // Separator WC.
+				$woocommerce_menu_order[] = $item; // Posts.
+				unset( $menu_order[ $orders ] );
+				unset( $menu_order[ $customers ] );
+				unset( $menu_order[ $products ] );
+				unset( $menu_order[ $analytics ] );
+				unset( $menu_order[ $marketing ] );
+				unset( $menu_order[ $extensions ] );
+				unset( $menu_order[ $woocommerce_separator ] );
+				unset( $menu_order[ $woocommerce_separator_top ] );
+
+				if ( false !== $woocommerce_payments ) {
+					unset( $menu_order[ $woocommerce_payments ] );
+				}
+
+			} elseif ( ! in_array( $item, array( 'wc-admin&path=/payments/connect', 'wc-calypso-bridge-separator-top', 'separator-woocommerce', 'woocommerce', 'woocommerce-marketing', 'wc-admin&path=/analytics/overview', 'edit.php?post_type=product', 'edit.php?post_type=shop_order', 'admin.php?page=wc-admin&path=/customers'), true ) ) {
```

Comment:
Nit: I think this would be easier to read using a multi-line array.
```suggestion
			} elseif ( ! in_array(
				$item,
				array(
					'wc-admin&path=/payments/connect',
					'wc-calypso-bridge-separator-top',
					'separator-woocommerce',
					'woocommerce',
					'woocommerce-marketing',
					'wc-admin&path=/analytics/overview',
					'edit.php?post_type=product',
					'edit.php?post_type=shop_order',
					'admin.php?page=wc-admin&path=/customers'
				),
				true
			) ) {
```